src/compose.py

```diff
@@ -3,24 +3,33 @@
 import versions
 
 
-def start(port, hostname):
-    run("up -d", port, hostname)
+def start(port, hostname, fake_db_annex):
+    run("up -d", port, hostname, fake_db_annex)
 
 
-def stop(port, hostname, persist_volumes):
+def stop(port, hostname, persist_volumes, fake_db_annex):
     if persist_volumes:
-        run("down", port, hostname)
+        run("down", port, hostname, fake_db_annex)
     else:
-        run("down --volumes", port, hostname)  # Also deletes volumes
+        run("down --volumes", port, hostname, fake_db_annex)  # Also deletes volumes
 
 
 def pull(port, hostname):
-    run("pull", port, hostname)
-
-
-def run(args, port, hostname):
-    args = "docker-compose --project-name montagu " + args
-    p = Popen(args, env=get_env(port, hostname), shell=True)
+    # NOTE: passing fake_db_annex = False here because it does not
+    # affect the pull (the fake db annex uses the main montagu-db
+    # container)
+    run("pull", port, hostname, False)
+
+
+def run(args, port, hostname, fake_db_annex):
+    prefix = 'docker-compose --project-name montagu '
+    if fake_db_annex:
+        # NOTE: it's surprising that the '../' is needed here, but
+        # docker-compose apparently looks like git through parent
+        # directories until it finds a docker-compose file!
+        prefix += '-f ../docker-compose.yml -f ../docker-compose-annex.yml '
```

Comment:
so if we're faking the annex, we include a container for the annex, but otherwise we don't? I'm a little confused by this

Reply:
You have it exactly right :)